### PR TITLE
Add discrete cosine transform code sample

### PR DESCRIPTION
**Pull request by ethanhirsch**
The windows ci configuration might need tweaking. I had issues testing it on my machine.

### CODE REVIEW

**Comment by anjgola** [comment]
@ethanhirsch  this is still in draft mode? Is that what you intended/

**Comment by ethanhirsch** [comment]
> @ethanhirsch this is still in draft mode? Is that what you intended/

Apologies, it should be ready now.

**Comment by ethanhirsch** [comment]
@JoeOster my Windows CI failed, I was wondering if there's some way to see the build logs?

**Comment by mkitez** [comment]
@ethanhirsch you can push a dummy commit to re-trigger CI. It will post the execution details in a comment.

**Comment by ethanhirsch** [comment]
> @ethanhirsch you can push a dummy commit to re-trigger CI. It will post the execution details in a comment.

@JoeOster Thanks Joe, I've gone ahead and done that but it appears to have invalidated your reviewer approval. Is there a way to retrigger the CI without it dismissing approvals for future reference?

**Comment by JoeOster** [comment]
Not that I know of

File: DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/DCT.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		{B2789FC1-D75F-4C6E-BF93-05BE084FBC79}.Intel Release|x64.ActiveCfg = Intel Release|x64
+		{B2789FC1-D75F-4C6E-BF93-05BE084FBC79}.Intel Release|x64.Build.0 = Intel Release|x64
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {55E32F12-32F5-4567-B736-8B38914AAB2E}
+	EndGlobalSection
+EndGlobal
```
Comment (anjgola) [inline DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/DCT.sln:25]: Should this file be merged? This seems like your personal workspace information.

File: DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/src/dpc_common.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#ifndef _DP_HPP
```
Comment (anjgola) [inline DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/src/dpc_common.hpp:1]: This should not be provided in sample source.

File: DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/DCT.sln (hunk 1 above)
Comment (ethanhirsch) [reply DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/DCT.sln:25]: Hey Anjali,

The .sln file is required for my command line build instructions documented in the README.

They require the user to call MSBuild and pass the .sln file as an argument, which allows the program to be build per the configurations set in visual studio.

As far as I know, this file shouldn't pose any issues, unless there's something I'm not aware of?


Side note: My latest commit should have resolved the dpc_common discrepancy. Thank you

Comment (anjgola) [reply DirectProgramming/DPC++/SpectralMethods/DiscreteCosineTransform/DCT.sln:25]: Okay, thanks for the explanation.

**Pull request merged**